firmware/source/mbFirmata.cpp

```diff
@@ -37,27 +37,13 @@ SOFTWARE.
 // scheduler, Firmata instantiates the individual DAL components it needs rather than using
 // the MicroBit object.
 
-MicroBitI2C i2c(I2C_SDA0, I2C_SCL0);
-MicroBitMessageBus messageBus;
-MicroBitSerial serial(USBTX, USBRX);
-MicroBitStorage storage;
-
-MicroBitAccelerometer &accelerometer = MicroBitAccelerometer::autoDetect(i2c);
-MicroBitButton buttonA(MICROBIT_PIN_BUTTON_A, MICROBIT_ID_BUTTON_A);
-MicroBitButton buttonB(MICROBIT_PIN_BUTTON_B, MICROBIT_ID_BUTTON_B);
-MicroBitCompass &compass = MicroBitCompass::autoDetect(i2c);
-MicroBitDisplay display;
-MicroBitThermometer thermometer(storage);
-
-MicroBitIO io(
-	MICROBIT_PIN_P0, MICROBIT_PIN_P1, MICROBIT_PIN_P2, MICROBIT_PIN_P3,
-	MICROBIT_PIN_P4, MICROBIT_PIN_P5, MICROBIT_PIN_P6, MICROBIT_PIN_P7,
-	MICROBIT_PIN_P8, MICROBIT_PIN_P9, MICROBIT_PIN_P10, MICROBIT_PIN_P11,
-	MICROBIT_PIN_P12, MICROBIT_PIN_P13, MICROBIT_PIN_P14, MICROBIT_PIN_P15,
-	MICROBIT_PIN_P16, /* 17-18 */ MICROBIT_PIN_P19, MICROBIT_PIN_P20);
+MicroBit uBit;
```

Comment:
I think the issue with using uBit here is that we'll run out of RAM on a V1?

Which component was the problem here?

It is no bad thing having the uBit component but @jhmaloney do you remember why you were using indivudual components? @microbit-sam does this still build for V1 like this?

Reply:
I don't remember, but I doubt this choice was driven by running out of RAM. It may have been simply a desire to use the minimal set of DAL features required. I have wanted to avoid competing with DAL processes for either CPU cycles or for certain hardware resources such as sensors.

You understand the internals of the DAL better than I do so if you think it makes sense to use uBit go for it. You can use the test suite to see if anything breaks. :-)